src/Compilers/Core/Portable/CodeGen/LocalSlotManager.cs

```diff
@@ -246,7 +252,54 @@ private LocalDefinition DeclareLocalImpl(
         internal void FreeSlot(LocalDefinition slot)
         {
             Debug.Assert(slot.Name == null);
-            FreeSlots.Push(new LocalSignature(slot.Type, slot.Constraints), slot);
+
+            if (_nonReusableLocals?.Contains(slot) != true)
```

Comment:
>_nonReusableLocals?.Contains(slot) != true [](http://example.com/codeflow?start=16&length=42)

Would it make sense to remove the entry from the hash table? Do we expect more than one attempt to free a local?

Reply:
> Do we expect more than one attempt to free a local?

No, as far as I can tell by investigating the callers of FreeSlot, that should never happen.

> Would it make sense to remove the entry from the hash table?

So I will do that. Thanks.